modules/cudaoptflow/src/nvidiaOpticalFlow.cpp

```diff
@@ -169,7 +201,8 @@ class NvidiaOpticalFlowImpl : public cv::cuda::NvidiaOpticalFlow_1_0
     NvOFHandle GetHandle() { return m_hOF; }
 
 protected:
-    HMODULE m_hModule; //module handle to load nvof dll
+    HMODULE m_hOFModule; //module handle to load nvof dll
+    HMODULE m_hCudaModule; //module handle to load cuda dll
```

Comment:
It should loaded once per process. Don't load .dll per class instance.

Reply:
Noted

Reply:
Please use singleton pattern for loading of required functions.

Reply:
Does my current changes address it?
I am loading it only when necessary

Reply:
> I am loading it only when necessary

Yes, you loading this on-demand.

> once

No, you loading libraries every time when class object is created.

Reply:
Okay Got it
So few doubts here @alalek 
1. We want a scenario where user can create multiple instances of same class. May be with bigger buffers and other stuff. In this case, we need to module handles global and load them only once. These global handles are within same namespace so is this approach OK?
There will be 2 global HMODULES 1 for cuda and 2 for opticalflow and one boolean to check if modules are loaded.

2. If i make it singleton then user will not be able to create mutiple instamces which blocks above usecase. Is ther any other way you suggest here?

Reply:
3rd variant should be straightforward.

You may skip library unloading process (a lot of code don't care about that and it may lead to unpredictable behavior during process termination).

[Example](https://github.com/opencv/opencv/blob/4.1.1/modules/videoio/src/cap_gstreamer.cpp#L184). Singleton class can provide pointers onto required functions (fields initialized in constructor).

Reply:
Is there any specific reason to skip library unloading part?. I am asking this since we faced some issues in other libraries due to not unloading.
I was thinking about having a refcount of class instances and when its 0, we can unload the libraries which are loaded and make them Null

Reply:
Process termination is tricky part on Win32: https://github.com/opencv/opencv/issues/12750
Especially working with asynchronous multi-threaded libraries like OpenCL or CUDA.
If unloading works, then - OK.

> having a refcount

Not needed. Also this would break "once" behavior again.
Unload once in singleton destructor (see example, add `printf` to debug that).

Reply:
Sure. Thanks :-)
Will add in destructor